syntax = "proto3";

package transport_catalogue_serialize;

message Distance {
    string from_stop = 1;
    string to_stop = 2;
    uint32 distance = 3;
}

message Stop {
    string stop_name = 1;
    double latitude = 2;
    double longitude = 3;
}

message Bus {
    string bus_number = 1;
    repeated string stops_at_route = 2;
    bool roundtrip = 3;
}


message Point {
    double x = 1;
    double y = 2;
}

message RenderSettings {
    double width = 1;
    double height = 2;
    
    double padding = 3;

    double line_width = 4;
    double stop_radius = 5;
    
    int32 bus_label_font_size = 6;
    Point bus_label_offset = 7;        // svg::Point

    double stop_label_font_size = 8;
    Point stop_label_offset = 9;       // svg::Point
    
    string underlayer_color = 10;
    double underlayer_width = 11;
    repeated string color_palette = 12;
}


message TransportCatalogue {
    repeated Stop stops = 1;
    repeated Bus buses = 2;
    repeated Distance distancies = 3;
    int32 bus_wait_time = 4;
    double bus_velocity = 5;
    RenderSettings render_settings = 6;
}